galaxy_ng/app/migrations/0017_populate_repos_and_remotes.py

```diff
@@ -0,0 +1,128 @@
+
+REPOSITORIES = [
+    {
+        "name": "published",
+        "description": "Certified content repository",
+        "pulp_type": "ansible.ansible",
+    },
+    {
+        "name": "community",
+        "description": "Community content repository",
+        "pulp_type": "ansible.ansible",
+    },
+    {
+        "name": "rh-certified",
+        "description": "Red Hat certified content repository",
+        "pulp_type": "ansible.ansible",
+    },
+]
+# NOTE:`staging` & `rejected` are created by 0010_add_staging_rejected_repos.py
+
+
+def populate_initial_repos(apps, schema_editor):
+    """Populates initial data for Automation Hub.
+
+    - Staging and rejected repos are covered by migration 0010
+    - Remotes are created only if doesn't exist and is specified in REMOTES dict
+      matching the same name as repository.
+    - Repositories are created using REPOSITORIES data and only if doesn't exist.
+    - Distributions are created only if not found with same name as repository.
+    - Reverse us NOOP so nothing is done if transaction fails to avoid corrupting
+      existing data.
+
+    """
+
+    db_alias = schema_editor.connection.alias
+
+    AnsibleRepository = apps.get_model('ansible', 'AnsibleRepository')
+    AnsibleRepository.objects = AnsibleRepository.objects.using(db_alias)
+
+    AnsibleDistribution = apps.get_model('ansible', 'AnsibleDistribution')
+    AnsibleDistribution.objects = AnsibleDistribution.objects.using(db_alias)
+
+    CollectionRemote = apps.get_model('ansible', 'CollectionRemote')
+    CollectionRemote.objects = CollectionRemote.objects.using(db_alias)
+
+    RepositoryVersion = apps.get_model('core', 'RepositoryVersion')
+    RepositoryVersion.objects = RepositoryVersion.objects.using(db_alias)
+
+    for repo_data in REPOSITORIES:
+
+        remote_data = REMOTES.get(repo_data["name"])
+        if remote_data is None:
+            remote = None
+        else:
+            remote = (
+                CollectionRemote.objects.filter(name=remote_data["name"]).first()
+                or
+                CollectionRemote.objects.create(**remote_data)
+            )
+
+        repository = (
+            AnsibleRepository.objects.filter(name=repo_data["name"]).first()
+            or
+            AnsibleRepository.objects.create(
+                remote=remote,
+                next_version=1,
+                **repo_data,
+            )
+        )
+
+        if not RepositoryVersion.objects.filter(repository__name=repository.name):
+            RepositoryVersion.objects.create(
+                repository=repository,
+                number=0,
+                complete=True
+            )
+
+        if not AnsibleDistribution.objects.filter(name=repository.name):
+            AnsibleDistribution.objects.create(
+                name=repository.name,
+                base_path=repository.name,
+                remote=remote,
+                repository=repository,
+                pulp_type="ansible.ansible",
```

Comment:
The model definition provides this field by default on the AnsibleDistribution subclass.

Reply:
@bmbouter I also added this after getting an error, I see other existing migrations also passing this field.

Reply:
Ok that'll work then. Ty for confirming it's needed.